### PR TITLE
Add Code Coverage Doc and some other drive-bys

### LINKED ISSUE #5723: Fix codecov.io Carry Forward Flags
#### Summary
The flags have been broken since we switched from e2e-ci to e2e-stable. We should do the following:

- Document the flags
- Ensure that stale flags do not carry forward

### PR DESCRIPTION
**Pull request by unlikelyzero**
Closes #5723 #5706 

### Describe your changes:

- [x] clean up lighthouse references
- [x] clean up code coverage flags
- [x] Fix up the couchdb github workflow
- [x] Doc improvements
- [x] Add couchdb tests to the nightly with coverage in circle ci
- [x] Fix test results not being uploaded on gha
- [x] document current code coverage process

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [ ] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/tests/functional/plugins/notebook/tags.e2e.spec.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
  *****************************************************************************/
 
 /*
-This test suite is dedicated to tests which verify form functionality.
+This test suite is dedicated to tests which verify notebook tag functionality.
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/tags.e2e.spec.js:24]: drive by

**Review by ozyx — COMMENTED** [review]
This all looks good to me. What remains to be documented of the current code coverage setup?

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Testing
+Open MCT Testing is iterating and improving at a rapid pace. This document serves to capture and index existing testing documentation and house documentation which no other obvious location as our testing evolves.
+
+## General Testing Process
+Documentation located [here](./docs/src/process/testing/plan.md)
+
+## Unit Testing
+Unit testing is essential part of our test strategy and compliments our e2e testing strategy.
```
Comment (akhenry) [inline TESTING.md:8]: compl**e**ments

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Codecov.io will combine each of the above commands with [Codecov.io Flags](https://docs.codecov.com/docs/flags). Effectively, this allows us to combine multiple reports which are run at various stages of our CI Pipeline or run as part of a parallel process.
+
+This e2e coverage is combined with our unit test report to give a comprehensive (if flawed) view of line coverage.
+
+### Limitations in our code coverage reporting
+
+Our code coverage implementation has two known limitations:
+- [Variability and accuracy](https://github.com/nasa/openmct/issues/5811)
+- []
```
Comment (akhenry) [inline TESTING.md:50]: This seems to be incomplete?

**Review by akhenry — DISMISSED** [review]
This is really nice cleanup, thank you for doing this. Just two small things.

File: TESTING.md (hunk 3 above)
Comment (unlikelyzero) [reply TESTING.md:50]: @akhenry documentation is incomplete but the rest is ready for review

**Review by ozyx — APPROVED** [review]

**Pull request merged**